pkg/manager/internal.go

```diff
@@ -260,6 +263,17 @@ func (cm *controllerManager) SetFields(i interface{}) error {
 	return nil
 }
 
+// AddMetricsExtraHandler adds extra handler served on path to the http server that serves metrics.
+func (cm *controllerManager) AddMetricsExtraHandler(path string, handler http.Handler) {
+	_, found := cm.metricsExtraHandlers[path]
+	if found {
+		panic(fmt.Errorf("can't register extra handler by duplicate path %q on metrics http server", path))
```

Comment:
Not sure if panic is preferred here over an explicit error, but I followed the way similar implemented in [webhook server](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/webhook/server.go#L110)

Reply:
I'd return an error as suggested above